go-controller/pkg/allocator/pod/pod_annotation_test.go

```diff
@@ -769,7 +822,7 @@ func Test_allocatePodAnnotationWithRollback(t *testing.T) {
 				g.Expect(podAnnotation.MAC[0]&2).To(gomega.BeEquivalentTo(2), "Expected local MAC")
 				return
 			}
-			g.Expect(podAnnotation).To(gomega.Equal(tt.wantPodAnnotation))
+			g.Expect(podAnnotation).To(gomega.Equal(tt.wantPodAnnotation), "diff: %s", cmp.Diff(tt.wantPodAnnotation, podAnnotation))
```

Comment:
is this really needed ? doesn't gomega print out the diffs between the expected and real results ?

Reply:
you are correct, however it adds the first compact diff form below, if you want i will remove it

```
W1222 14:44:10.132065   24043 pod_annotation.go:368] failed to ensure requested or annotated IPs [192.168.0.4/24] for default/namespace/pod: Allocate IPs failed
E1222 14:44:10.132381   24043 pod_annotation.go:316] Pod namespace/pod referencing an IPAMClaim on network "network" which does not honor it
--- FAIL: Test_allocatePodAnnotationWithRollback (0.00s)
    --- FAIL: Test_allocatePodAnnotationWithRollback/expect_new_IP_and_ipv6_gateway_LLA_for_primary_udn_layer2_with_dual_stack (0.00s)
        pod_annotation_test.go:825: 
            diff:   &util.PodAnnotation{
              	IPs: []*net.IPNet{
              		&{IP: s"192.168.0.3", Mask: s"ffffff00"},
              		&{
            - 			IP:   s"200:100:200::3",
            + 			IP:   s"2010:100:200::3",
              			Mask: s"fffffffffffffff00000000000000000",
              		},
              	},
              	MAC:      s"0a:58:c0:a8:00:03",
              	Gateways: {s"192.168.0.1", s"2010:100:200::1"},
              	... // 4 identical fields
              }
            
            Expected
                <*util.PodAnnotation | 0xc00041b0e0>: {
                    IPs: [
                        {
                            IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 192, 168, 0, 3],
                            Mask: [255, 255, 255, 0],
                        },
                        {
                            IP: [32, 16, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3],
                            Mask: [255, 255, 255, 255, 255, 255, 255, 240, 0, 0, 0, 0, 0, 0, 0, 0],
                        },
                    ],
                    MAC: [10, 88, 192, 168, 0, 3],
                    Gateways: [
                        [192, 168, 0, 1],
                        [32, 16, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
                    ],
                    IPv6LLAGateway: [254, 128, 0, 0, 0, 0, 0, 0, 8, 88, 100, 255, 254, 65, 0, 4],
                    Routes: [
                        {
                            Dest: {
                                IP: [100, 65, 0, 0],
                                Mask: [255, 255, 0, 0],
                            },
                            NextHop: [192, 168, 0, 1],
                        },
                        {
                            Dest: {
                                IP: [253, 153, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
                                Mask: [255, 255, 255, 255, 255, 255, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0],
                            },
                            NextHop: [32, 16, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
                        },
                    ],
                    TunnelID: 100,
                    Role: "primary",
                }
            to equal
                <*util.PodAnnotation | 0xc0001d7b90>: {
                    IPs: [
                        {
                            IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 192, 168, 0, 3],
                            Mask: [255, 255, 255, 0],
                        },
                        {
                            IP: [2, 0, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3],
                            Mask: [255, 255, 255, 255, 255, 255, 255, 240, 0, 0, 0, 0, 0, 0, 0, 0],
                        },
                    ],
                    MAC: [10, 88, 192, 168, 0, 3],
                    Gateways: [
                        [192, 168, 0, 1],
                        [32, 16, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
                    ],
                    IPv6LLAGateway: [254, 128, 0, 0, 0, 0, 0, 0, 8, 88, 100, 255, 254, 65, 0, 4],
                    Routes: [
                        {
                            Dest: {
                                IP: [100, 65, 0, 0],
                                Mask: [255, 255, 0, 0],
                            },
                            NextHop: [192, 168, 0, 1],
                        },
                        {
                            Dest: {
                                IP: [253, 153, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
                                Mask: [255, 255, 255, 255, 255, 255, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0],
                            },
                            NextHop: [32, 16, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
                        },
                    ],
                    TunnelID: 100,
                    Role: "primary",
                }
FAIL
exit status 1
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/pod	0.015s
```

Reply:
I mostly worry that we're not doing this anywhere else ... Are we ?

Reply:
OVN doesnt, OVN/vendor does fwiw
imho it is nit, and good to have
up to you